react/index.tsx

```diff
@@ -0,0 +1,141 @@
+import React, {
+  Fragment,
+  FunctionComponent,
+  ReactElement,
+  useState,
+  useCallback,
+} from 'react'
+import classNames from 'classnames'
+import { generateBlockClass, BlockClass } from '@vtex/css-handles'
+//@ts-ignore
+import { Modal } from 'vtex.styleguide'
+//@ts-ignore
+import styles from './styles.css'
+
+interface ModalProps {
+  centered: boolean
+  closeOnEsc: boolean
+  closeOnOverlayClick: boolean
+  showCloseIcon: boolean
+  children: ReactElement
+  buttonLabel: string
+  buttonClass: string | null
+}
+
+interface StorefrontComponent
+  extends FunctionComponent<ModalProps & BlockClass> {
+  schema?: any
+}
+
+const ModalComponent: StorefrontComponent = ({
+  centered,
+  closeOnEsc,
+  closeOnOverlayClick,
+  showCloseIcon,
+  children,
+  buttonLabel,
+  buttonClass,
+  blockClass,
+}) => {
+  const [isOpen, setOpen] = useState(false)
+  const onClick = useCallback(() => setOpen(true), [])
+  const onClose = useCallback(() => setOpen(false), [])
+
+  const buttonClasses = classNames(
+    `${generateBlockClass(
+      styles.button,
+      blockClass
+    )} t-action pointer pv3 nv3 ph4 nh4 c-action-primary`,
+    {
+      [`${buttonClass}`]: !!buttonClass,
```

Comment:
You can just pass `buttonClass` to `classNames` and if `buttonClass`  is falsy, it won't be applied.